langroid/vector_store/weaviatedb.py

```
@@ -0,0 +1,266 @@
+import logging
+import os
+from typing import Dict, List, Optional, Sequence, Tuple, TypeVar
+
+from dotenv import load_dotenv
+
+from langroid.embedding_models.base import (
+    EmbeddingModelsConfig,
+)
+from langroid.embedding_models.models import OpenAIEmbeddingsConfig
+from langroid.exceptions import LangroidImportError
+from langroid.mytypes import DocMetaData, Document, EmbeddingFunction, Embeddings
+from langroid.utils.configuration import settings
+from langroid.vector_store.base import VectorStore, VectorStoreConfig
+
+logger = logging.getLogger(__name__)
+try:
+    import weaviate
+    from weaviate.classes.config import (
+        Configure,
+        VectorDistances,
+    )
+    from weaviate.classes.init import Auth
+    from weaviate.classes.query import Filter, MetadataQuery
+    from weaviate.util import generate_uuid5, get_valid_uuid
+except ImportError:
+    raise LangroidImportError("weaviate", "weaviate")
+
+
+class WeaviateDBConfig(VectorStoreConfig):
+    cloud: bool = True
+    collection_name: str | None = "temp"
+    storage_path: str = ".weaviate/data"
+    embedding: EmbeddingModelsConfig = OpenAIEmbeddingsConfig()
+    distance: str = VectorDistances.COSINE
+
+
+class WeaviateDB(VectorStore):
+    def __init__(self, config: WeaviateDBConfig = WeaviateDBConfig()):
+        super().__init__(config)
+        self.config: WeaviateDBConfig = config
+        self.embedding_fn: EmbeddingFunction = self.embedding_model.embedding_fn()
+        self.embedding_dim = self.embedding_model.embedding_dims
+        load_dotenv()
+        key = os.getenv("WEAVIATE_API_KEY")
+        url = os.getenv("WEAVIATE_API_URL")
+        if config.cloud and None in [key, url]:
```

related to above comment about not needing `config.cloud`.
If we're not using this flag, then we should raise an error if key or url is None.